CMakeLists.txt

```diff
@@ -145,7 +146,16 @@ find_package(PkgConfig REQUIRED)
 #TODO: proper find module for SDL2
 pkg_check_modules(SDL2 REQUIRED IMPORTED_TARGET sdl2)
 set(OpenGL_GL_PREFERENCE "GLVND")
-find_package(OpenGL REQUIRED)
+
+find_package(OpenGL QUIET)
+if(OpenGL_FOUND)
+	message(STATUS "Using OpenGL GLX for OpenGL")
+else(OpenGL_FOUND)
+	# If we couldn't find old GLX package, try to use modern libglvnd libOpenGL.so instead.
+	message(STATUS "Using GLVND for OpenGL")
+	FIND_LIBRARY(OpenGL_LIBRARY OpenGL)
+endif(OpenGL_FOUND)
+
```

Comment:
Hi! Let's begin with this one.
On my system, I see this without this change:```


~/src/g2/gish/b4# cmake ..
-- The C compiler identification is GNU 10.2.1
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Found PkgConfig: /usr/bin/pkg-config (found version "0.29.2") 
-- Checking for module 'sdl2'
--   Found sdl2, version 2.0.22
CMake Error at /usr/share/cmake-3.18/Modules/FindPackageHandleStandardArgs.cmake:165 (message):
  Could NOT find OpenGL (missing: OPENGL_glx_LIBRARY)
Call Stack (most recent call first):
  /usr/share/cmake-3.18/Modules/FindPackageHandleStandardArgs.cmake:458 (_FPHSA_FAILURE_MESSAGE)
  /usr/share/cmake-3.18/Modules/FindOpenGL.cmake:433 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
  CMakeLists.txt:148 (find_package)


-- Configuring incomplete, errors occurred!
See also "/root/src/g2/gish/b4/CMakeFiles/CMakeOutput.log".
```

So that is what this code fixes.
I don't even have GLX on my systems. I don't have anything X11 installed, in fact.
So, it seems that GLVND (which is installed and in good use) is not found.
Is there a different approach to try? What do you mean with `FindOpenGL` in your comment?

In a nutshell, doing this on Debian 11 with latest MESA stable version built from sources (MESA 22.1.3), GLVND is not found:
```

set(OpenGL_GL_PREFERENCE "GLVND")
find_package(OpenGL REQUIRED)
```

CMake version on Debian 11 STABLE is 3.16.